### PR TITLE
[Serializer] Update serializer example using type hints

### PR DESCRIPTION
**Pull request by 94noni**
Hi, small proposition to upgrade code example here in the serializer doc using type hint on object to serialize to make example more robust from the input to output

Before continuing on this page, I want to ask if it is useful or not
Thank you!

### CODE REVIEW

**Comment by javiereguiluz** [comment]
I like this proposal. @wouterj do you agree on adding this in 6.0 branch and up? Thanks!

**Comment by javiereguiluz** [comment]
My vote is 👍  ... but please only update this article and not all docs but that would create multiple hard-to-solve merge conflicts. Thanks!

But first, let's wait to read the opinion of @xabbuh and @OskarStark. Thanks!

File: components/serializer.rst [hunk 1]
```diff
@@ -96,28 +96,28 @@ exists in your project::
...
 
         // Setters
-        public function setName($name)
+        public function setName(string $name): void
         {
             $this->name = $name;
         }
 
-        public function setAge($age)
+        public function setAge(int $age): void
```
Comment (OskarStark) [inline components/serializer.rst:110]: If we change the order above, we should move setAge() above setName()

**Comment by wouterj** [comment]
:+1: for 4.4 and up. This is what we have done in most other cases and will result in the least merge difficulty.

(Javier, if you want me to take care of (up-)merging this let me know)

**Comment by javiereguiluz** [comment]
Antoine, thanks for this improvement. Merged!

File: components/serializer.rst (hunk 1 above)
Comment (javiereguiluz) [reply components/serializer.rst:110]: I changed the _setter_ order while merging. Thanks.

**Comment by 94noni** [comment]
@javiereguiluz @wouterj thx for following this PR and merging it

would it be interesting/valuable to create issue and tag them like for translations as help wanted or easy to pick ?
pointing to this PR as reference of code how to :)

**Pull request merged**